File: 1481. Least Number of Unique Integers after K Removals/findLeastNumOfUniqueInts.cpp
class Solution
{
public:
    int findLeastNumOfUniqueInts(vector<int> &arr, int k)
    {

        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> minh;
        unordered_map<int, int> m;
        for (auto it : arr)
        {
            m[it]++;
        }

        for (auto it : m)
        {
            minh.push({it.second, it.first});
        }

        while (minh.size() > 0)
        {
            if (minh.top().first <= k)
            {
                k -= minh.top().first;
                minh.pop();
            }
            else if (minh.top().first > k)
            {
                break;
            }
        }

        return minh.size();
    }
};

File: 1481. Least Number of Unique Integers after K Removals/findLeastNumOfUniqueInts.java
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();

        for (int num : arr) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        PriorityQueue<Integer> minHeap = new PriorityQueue<>(Comparator.comparingInt(frequencyMap::get));

        for (int key : frequencyMap.keySet()) {
            minHeap.offer(key);
        }

        while (!minHeap.isEmpty() && k > 0) {
            int num = minHeap.poll();
            k -= frequencyMap.get(num);
            if (k >= 0) {
                frequencyMap.remove(num);
            }
        }

        return frequencyMap.size();
    }
}